internal/runners/exec/exec.go

```diff
@@ -104,7 +109,12 @@ func (s *Exec) Run(params *Params, args ...string) error {
 	logging.Debug("Trying to exec %s on PATH=%s", args[0], env["PATH"])
 	// Ensure that we are not calling the exec recursively
 	if _, isBeingShimmed := env[constants.ExecEnvVarName]; isBeingShimmed {
```

Comment:
We should just get rid of the env var, since the new detection you implemented doesn't depend on it.